src/context.py

```diff
@@ -1,7 +1,75 @@
+from datetime import datetime
+
 import discord
+
 from client import client
 from config import MAILBOX
-from commonbot.utils import combine_message, send_message
+from commonbot.utils import combine_message
+from forwarder import message_forwarder
+
+class ReportResolveButton(discord.ui.Button):
+    def __init__(self):
+        super().__init__(
+            label="Resolve",
+            style=discord.ButtonStyle.success
+        )
+
+    async def callback(self, interaction: discord.Interaction):
+        embed: discord.Embed = interaction.message.embeds[0]
+        embed.colour = discord.Colour.dark_green()
+        embed.title = f"\N{WHITE HEAVY CHECK MARK}{embed.title[1:]}"
+        embed.add_field(
+            name=f"Resolved <t:{int(datetime.now().timestamp())}:R> by",
+            value=interaction.user.mention,
+            inline=False
+        )
+
+        view: discord.ui.View = self.view
+        view.remove_item(self)
+        await interaction.message.edit(embed=embed, view=view)
+        await interaction.response.defer()
+
+
+class ReportThreadButton(discord.ui.Button):
+    def __init__(self, *, thread_url: str = None, reported_user: discord.User = None):
+        super().__init__(
+            label="Thread",
+            style=discord.ButtonStyle.link if thread_url else discord.ButtonStyle.secondary,
+            emoji=None if thread_url else "\N{LEFT-POINTING MAGNIFYING GLASS}",
+            url=thread_url
+        )
+        self.reported_user: discord.User = reported_user
+
+    async def callback(self, interaction: discord.Interaction):
+        if self.url is None:
+            log_chan = interaction.guild.get_channel(MAILBOX)
+            thread: discord.Thread = await message_forwarder.get_or_create_user_reply_thread(
+                user=self.reported_user,
+                parent_channel=log_chan
```

Comment:
This function doesn't have a `parent_channel` parameter?

Reply:
It did when I committed 🌧